src/main/java/school/hei/haapi/service/UserService.java

```diff
@@ -153,6 +154,28 @@ public List<User> getByRoleAndStatus(User.Role role, User.Status status) {
     return userRepository.findAllByRoleAndStatus(role, status);
   }
 
+  public <T> byte[] getByRoleAndStatusAsXlsx(
+      User.Role role, User.Status status, Function<User, T> mapper) {
+    List<User> users = getByRoleAndStatus(role, status);
+    XlsxCellsGenerator<T> generator = new XlsxCellsGenerator<>();
+    List<T> mappedUsers = users.stream().map(mapper).toList();
```

Comment:
Where is your test ? I think you can directly use the STAFF_MEMBER User type here, also don't forget to test your code before merging. 

How we will work:
.1 write the api.doc (indeed)
.2 if all are ok we will merge the spec and generate the client. Then Frontend can do their work part
.3 and it's after all that we will implement

Reply:
+ Test

Reply:
Understood

Reply:
Tests are now done